wireless_modules/sensors/sensor_base.py

```diff
@@ -0,0 +1,11 @@
+"""
+Abstract base class for all sensors
+"""
+
+
+class Sensor:
```

Comment:
Make this an actual ABC by doing:

```
from abc import ABC, abstractmethod

class Sensor(ABC):
    def __init__(self):
        pass

    @abstractmethod
    def read(self):
        pass
```

PS: Don't forget docstrings!

Reply:
Where's my syntax highlighting, I can't read this!

Reply:
Woah I didn't know Python needed this 😅. And what doc string comment would be needed here? Like explain why the read method is an abstract method?

Reply:
If you're talking about docstrings for abstract _classes_, then I think it should just be the same as explaining what a class does. For me a good checklist would be:
- Say its an abstract class (so developers know it needs to be extended)
- Summarise what the class does or represents
- List important class/instance variables that are shared between its subclasses (in this case, none)
- How its meant to be implemented or how it should be used

But if you're talking about writing docstrings for an abstract _method_, then it should be fine to just write what it should do for its subclasses. Or don't and just write `pass`, that's fine too, because you can always write the docstring in the subclasses.

The following pics are from `data.py` in the Raspicam repo that I've been working on. I used these because the files were open for me at the time 😆 , but hopefully these can be good examples. 
![image](https://user-images.githubusercontent.com/58575113/92998902-2256c980-f560-11ea-88b6-571da56bb1a2.png)

![image](https://user-images.githubusercontent.com/58575113/92998911-2f73b880-f560-11ea-926f-a0769c544b2f.png)

Reply:
Double-backed and saw what you currently have for `Sensor` and I think this works fine.

Reply:
Thanks a lot for the detailed explanation @nix-chu! I was more confused because the sensor base class as it is didn't need much explaining with comments other than (in my opinion) explaining how the read method should be implemented. But I like the checklist you summarised above, it's quite handy!